public/app/services/searchService.js

```diff
@@ -11,10 +11,12 @@
                 var queryString;
 
                 queryString = function (parameters) {
-                    return _.map(parameters, function (value, key) {
-                        return '(' + key + ':' + value + ')';
+                    return _.map(parameters.children, function (child) {
+                        return child.hasOwnProperty('children') ?
+                            '(' + queryString(child) + ')' :
+                            '(' + child.field + ':' + child.value + ')'; // TODO Use comparator
```

Comment:
Think I'll merge as-is as it gets things deployed quickly for the period you are away. Will add this into YouTrack to be performed at a later point.